ui/src/app/views/notebook-list/component.ts

```diff
@@ -158,6 +166,59 @@ export class NotebookListComponent implements OnInit, OnDestroy {
     this.openNotebook();
   }
 
+  renameThis(notebook: String): void {
+    this.notebookInFocus = notebook;
+    this.renameModal.open();
+  }
+
+  receiveRename($event): void {
+    this.renameNotebook($event);
+  }
+
+  renameNotebook(rename): void {
+    let newName = rename.newName;
+    if (!(new RegExp('^.+\.ipynb$').test(newName))) {
+      newName = rename.newName + '.ipynb';
+    }
+    if (this.notebookList.filter((nb) => nb.name === newName).length > 0) {
+      this.renameModal.close();
+      this.notebookRenameConflictError = true;
+      this.duplicateName = newName;
+      return;
+    }
+   this.workspacesService.renameNotebook(this.wsNamespace, this.wsId, rename).subscribe(() => {
```

Comment:
Can we make this return the updated `FileDetail` for display? Just to follow the same 'update' semantics as the rest of the API?